modules/apps/powwow/powwow-portlet/docroot/WEB-INF/src/com/liferay/powwow/meetings/portlet/MeetingsPortlet.java

```diff
@@ -1243,6 +1354,106 @@ private Calendar _toJCalendar(DateValue dateValue, TimeZone timeZone) {
+			PowwowMeetingOccurrence powwowMeetingOccurrence,
+				CalendarBooking calendarBooking, long[] childCalendarIds,
+					Map<Locale, String> titleMap, Map<Locale, String> descriptionMap,
+						ThemeDisplay themeDisplay, ServiceContext serviceContext) throws Exception {
+
+		Recurrence recurrence = calendarBooking.getRecurrenceObj();
+
+		String zoomOriginalData = powwowMeetingOccurrence.getZoomOriginalData();
+		JSONObject zoomOriginalDataJSON = JSONFactoryUtil.createJSONObject(zoomOriginalData);
+
+		Calendar startTimeOriginalCalendar =
+			_parseZoomUtcTime(
+				zoomOriginalDataJSON.getString("start_time"),
+					calendarBooking.getTimeZone());
+
+		recurrence.addExceptionDate(startTimeOriginalCalendar);
+
+		String recurrenceSerializedData = RecurrenceSerializer.serialize(recurrence);
+
+		CalendarBookingLocalServiceUtil.updateCalendarBooking(
+			themeDisplay.getUserId(),
+			calendarBooking.getCalendarBookingId(),
+			calendarBooking.getCalendarId(), childCalendarIds, titleMap,
+			descriptionMap, StringPool.BLANK, calendarBooking.getStartTime(),
+			calendarBooking.getEndTime(), false,
+			recurrenceSerializedData, 0, "email", 0, "email", serviceContext);
+	}
+
+	private void _updateCalendarBookingOccurrence(
+		PowwowMeeting powwowMeeting,
+			PowwowMeetingOccurrence powwowMeetingOccurrence,
+				long startTime, long endTime, long[] childCalendarIds,
+					Map<Locale, String> titleMap, Map<Locale, String> descriptionMap,
+						ThemeDisplay themeDisplay, ServiceContext serviceContext) throws Exception {
+
+		long calendarBookingId = 0;
+		long occurrenceId = powwowMeetingOccurrence.getOccurrenceId();
+
+		if (powwowMeetingOccurrence.getCalendarBookingId() > 0) {
+			CalendarBooking calendarBookingOccurrence =
+				CalendarBookingLocalServiceUtil.fetchCalendarBooking(
+					powwowMeetingOccurrence.getCalendarBookingId());
+
+			CalendarBookingLocalServiceUtil.updateCalendarBooking(
+				themeDisplay.getUserId(),
+				calendarBookingOccurrence.getCalendarBookingId(),
+				calendarBookingOccurrence.getCalendarId(), childCalendarIds, titleMap,
+				descriptionMap, StringPool.BLANK, startTime, endTime, false,
+				StringPool.BLANK, 0, "email", 0, "email", serviceContext);
+
+		}
+		else {
+			long calendarId = getCalendarId(themeDisplay.getUserId(), serviceContext);
+
+			CalendarBooking calendarBookingNew =
+				CalendarBookingLocalServiceUtil.addCalendarBooking(
+					themeDisplay.getUserId(), calendarId, childCalendarIds ,
+					CalendarBookingConstants.PARENT_CALENDAR_BOOKING_ID_DEFAULT,
+					titleMap, descriptionMap, StringPool.BLANK, startTime, endTime,
+					false, StringPool.BLANK, 0, "email", 0, "email",
+					serviceContext);
+
+			calendarBookingId = calendarBookingNew.getCalendarBookingId();
+		}
+
+		PowwowMeetingOccurrenceLocalServiceUtil
+			.updatePowwowMeetingOccurrence(
+				occurrenceId, calendarBookingId, startTime, endTime,
+					OccurrenceStatus.parse(powwowMeetingOccurrence.getOccurrenceStatus()));
```

Comment:
i already exposed a method to return enum `owwowMeetingOccurrence.getOccurrenceStatusEnum()`